ompi/mca/pml/ucx/pml_ucx_request.h

```diff
@@ -99,7 +99,7 @@ struct pml_ucx_persistent_request {
     unsigned                          flags;
     void                              *buffer;
     size_t                            count;
-    union {
+    struct {
```

Comment:
no need struct now, just make ucp_datatype and ompi_datatype part of persistent_request